src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -537,70 +547,136 @@ public static class SystemNamespaceType
                     }
                 }
 
-                return LanguageConstants.String;
+                return new(LanguageConstants.String);
             };
 
-        private static TypeSymbol LoadTextContentTypeBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
+        private static FunctionResult LoadTextContentResultBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
         {
-            if (argumentTypes[0] is not StringLiteralType filePathType)
+            return TryLoadTextContentFromFile(binder, fileResolver, diagnostics,
```

Comment:
Is there any caching involved with reading these files? If not, presumably we'll have to reload each file from disk on every keypress. Are there any performance concerns here?

Reply:
No, there isn't. This was like that since initial implementation of load* functions. We decided not to overcomplicate as files were quite limited and there was no significant decrease in performance observed.

Reply:
That's fine - I'll raise an issue just so we have it tracked, but I agree no need to overcomplicate things right now unless we know it's going to be a problem.